lib/api/web-element/scoped-element.js

```diff
@@ -180,7 +180,10 @@ class ScopedWebElement {
 
         return await this.findElement({parentElement, selector: condition, index, timeout, retryInterval});
       } catch (error) {
-        if (this.suppressNotFoundErrors) {
+        if (
+          this.suppressNotFoundErrors &&
+          error.name === 'NoSuchElementError'
+        ) {
           return null;
         }
```

Comment:
@Vaibhavsahu2810 Thanks for this PR. You're going in the right direction with this but your solution is still incomplete and it seems that it's not tested properly against the valid cases.

For example, if you run `await browser.element.find({selector: '.something', suppressNotFoundErrors: true});` with your solution, it would still result in the `NoSuchElementError` error while such errors should ideally have been suppressed. Can you find out the reason for that and fix it?

Reply:
@garg3133 This is happening because `TimeoutError` was being converted into `NoSuchElementError` after the check for `NoSuchElementError`. 
Specifically:

1. If the element search timed out, the error was initially raised as a `TimeoutError`.

2. The `createNarrowedError` function then converted this `TimeoutError` into a `NoSuchElementError` for clarity.

However, the check for `suppressNotFoundErrors` was occurring before this conversion.That's why during the check error.name is TimeoutError

Reply:
So by checking,
```
if ( this.suppressNotFoundErrors && (error.name === "TimeoutError" || error.name === `"NoSuchElementError"))` 
```
we can suppress `NoSuchElementError` error

Reply:
Or, you could move the `createNarrowedError()` function call to the top of the catch block (just that particular line) and just have the check for the `NoSuchElementError`.